### PR TITLE
Waf resource import

### PR DESCRIPTION
**Pull request by oscarDovao**
This PR contains a new waf version struct method that checks for emptiness. Empty, in this case, means that non of the configuration fields are populated. This method can be used to prevent sending a back-end server request which will result on an error

### CODE REVIEW

**Comment by oscarDovao** [comment]
Thanks @phamann for the detailed explanation.

I see how using pointers would make the update more accurate and flexible in general terms.
Looking again at the resource's schema, i see a number of `TypeInt` fields which could technically (regardless the practicality, but for consistency) be set as 0 value and that would fail to be reflected on the server if the default is other than 0.

I like the Optional pattern and the semantics it brings to the implementation, but I agree it would be a bigger change for go-fastly SDK users to adopt. 

Given that the current implementation of the WAF configuration doesn't work for 100% of the cases, we'd like to change this PATCH endpoint's to use pointers and document the reason why this is different from the others endpoints. We can have further discussions on making a bigger change so this issue is addressed across the entire SDK.

Please let us know if you agree with the change 
Thanks

File: fastly/waf_version.go [hunk 1]
```diff
@@ -226,60 +226,72 @@ func (c *Client) GetWAFVersion(i *GetWAFVersionInput) (*WAFVersion, error) {
 	return &wafVer, nil
 }
 
-// UpdateWAFVersionInput is used as input to the UpdateWAFVersion function.
+// UpdateWAFVersionInput is used as input to the UpdateWAFVersion function. This struct uses pointers due to the problem
+// detailed on this issue https://trello.com/c/dbF8uqKn.
```
Comment (pegerto) [inline fastly/waf_version.go]: happy to have this in trello, as this is public probably make sense to reported as a GitHub issue (enhancement)

File: fastly/waf_version_test.go [hunk 2]
```diff
@@ -160,92 +160,92 @@ func TestClient_WAF_Versions(t *testing.T) {
 
 func verifyWAFVersionUpdate(t *testing.T, i *UpdateWAFVersionInput, o *WAFVersion) {
 
-	if i.WAFVersionID != o.ID {
-		t.Errorf("expected %s waf: got %s", i.WAFVersionID, o.ID)
+	if *i.WAFVersionID != o.ID {
```
Comment (pegerto) [inline fastly/waf_version_test.go]: [IDEA] use testify to manage assertions,

File: fastly/waf_version.go (hunk 1 above)
Comment (phamann) [inline fastly/waf_version.go]: Can we please remove the reference to Trello and replace with a link to the public GitHub PR instead.

Comment (oscarDovao) [reply fastly/waf_version.go]: Changed

Comment (oscarDovao) [reply fastly/waf_version.go]: Used the PR link

File: fastly/waf_version_test.go (hunk 2 above)
Comment (oscarDovao) [reply fastly/waf_version_test.go]: I agree, but probably a bigger change so that it is used it consistently across all tests. @phamann any thoughts?

Comment (phamann) [reply fastly/waf_version_test.go]: I would prefer not to introduce that change in this pr.

**Pull request merged**